### ISSUE TITLE
Block Comments Accessibility: Options button is focused, Some element in the sidebar should be focused

### ISSUE THREAD

**Issue by jeffpaul** [WordPress/gutenberg#71676]
### Description

From @t-hamano in https://github.com/WordPress/gutenberg/issues/71249#issue-3333938703:

> Click the Comment button in the block toolbar:
> The Options button is focused. Some element in the sidebar should be focused.

From @joedolson in https://github.com/WordPress/gutenberg/issues/71249#issuecomment-3201284909:

> There's no mechanism to navigate back to the block you commented on.
> There should be a link in the comment area where you can return to the selected block. Otherwise, it's going to be very labor intensive for a keyboard user to review comments.

### Step-by-step reproduction instructions

.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by t-hamano** [comment]
I think #71308 doesn't address the issue reported here. #71308 just scrolls to the relevant block when a comment is selected and visually highlights it.

**Comment by roseg43** [comment]
@jeffpaul Noting that this should be moved back to Todo since as @t-hamano pointed out, at least part of this issue hasn't been resolved. Specifically, @joedolson's feedback:

> There's no mechanism to navigate back to the block you commented on.
There should be a link in the comment area where you can return to the selected block. Otherwise, it's going to be very labor intensive for a keyboard user to review comments.
